### ISSUE TITLE
Example files do not work

### ISSUE THREAD

**Issue by Exegetech** [eprbell/rp2#10]
Hi, I'm trying to run the example files:
- crypto_example.config
- crypto_example.ods

But upon running, get these errors

```
INFO: Country: us
INFO: Accounting Method: fifo
INFO: Configuration file: crypto_example.config
INFO: Input file: crypto_example.ods
INFO: Processing BTC
INFO: Processing ETH
ERROR: Fatal exception occurred:
Traceback (most recent call last):
  File "/Users/swissarmybox/Desktop/cry/rp2stuff/venv/lib/python3.8/site-packages/rp2/rp2_main.py", line 91, in _rp2_main_internal
    input_data: InputData = parse_ods(configuration=configuration, asset=asset, input_file_handle=input_file_handle)
  File "/Users/swissarmybox/Desktop/cry/rp2stuff/venv/lib/python3.8/site-packages/rp2/ods_parser.py", line 123, in parse_ods
    transaction = _create_transaction(configuration, current_table_type, i + 1, row_values)
  File "/Users/swissarmybox/Desktop/cry/rp2stuff/venv/lib/python3.8/site-packages/rp2/ods_parser.py", line 190, in _create_transaction
    argument_pack: Dict[str, Any] = configuration.get_in_table_constructor_argument_pack(row_values)
  File "/Users/swissarmybox/Desktop/cry/rp2stuff/venv/lib/python3.8/site-packages/rp2/configuration.py", line 163, in get_in_table_constructor_argument_pack
    return self.__get_table_constructor_argument_pack(data, "in", self.__in_header)
  File "/Users/swissarmybox/Desktop/cry/rp2stuff/venv/lib/python3.8/site-packages/rp2/configuration.py", line 154, in __get_table_constructor_argument_pack
    raise RP2ValueError(
rp2.rp2_error.RP2ValueError: Parameter 'data' has length 13, but required minimum from in-table headers in crypto_example.config is 14: ['2020-06-03T11:23Z', 'Coinbase', 'Bob', 'brown', 'fox', 'Buy', 'ETH', 10.0, 244.0, 2440.0, 2465.0, 25.0, None]
INFO: Log file: ./log/rp2_2022_02_26_09_58_40_020721.log
INFO: Generated output directory: output
INFO: Done
```

**Comment by swissarmybox** [comment]
Sorry I have some questions regarding the example file. I don't know if I should open a new issue or I should just describe it here.

In the generated full report file, BTC In-Out, row 24, 26, 27, why is it a taxable event? Is it a taxable event to Bob?
Same goes with row 16, 17, why is it a taxable event? Is it a taxable event to Bob?

**Comment by eprbell** [comment]
No worries, here is fine, since it's still related to crypto_example. 

Row 16 and 17 are a gift and a donation by Bob, which are taxable events (see fields highlighted in yellow for the fiat amount of the event). They are reported in the Donations and Gifts tab in the tax_report_us.ods output file. 

Rows 24, 26, 27 refer to the fees for moving funds from one account to another (again, see highlighted field). Technically these fees are similar to very small sales and they do affect the in/out relationships among lots, so RP2 captures them in the full report and groups them together in the "Investment Expenses" tab in the tax_report_us.ods output file. I'm not 100% sure if transfer fees are taxable or not (see also [this relevant FAQ](https://github.com/eprbell/rp2/blob/main/docs/user_faq.md#if-i-transfer-cryptocurrency-between-two-accounts-i-own-is-the-fee-taxable): you can either err on the side of safety and treat them as sales or double check with a tax professional (show them the "Investment Expenses" tab in tax_report_us.ods).

Hope this helps.

**Comment by swissarmybox** [comment]
> 

Thanks, yeah definitely helps. I'll study more of the examples and will probably ask more questions here. Thanks a lot.

**Comment by swissarmybox** [comment]
How do we handle crypto loss? i.e, from scam or from forgetting keys?

**Comment by eprbell** [comment]
Thanks for all the good questions and feedback: this will be useful for others too.

Here's my understanding on the topic of theft, displacement and rug pulls:
* crypto theft/displacement is not tax deductible (therefore it doesn't need a transaction type in RP2)
* rug pulls are treated as a normal loss (so again, no need for a transaction type in RP2)

Here are articles that discuss this topic: 
* https://www.blog.cointracking.info/do-you-pay-taxes-stolen-hacked-lost-crypto
* https://cryptotrader.tax/blog/reporting-stolen-or-lost-cryptocurrency-for-tax-purposes

From the perspective of RP2 the only question is: is a new transaction type required? I think the answer is no (existing ones are sufficient to capture these scenarios), but I'm happy to stand corrected if anybody has a better understanding.

**Comment by swissarmybox** [comment]
Gotcha, that makes sense. Thanks.

**Comment by swissarmybox** [comment]
I looked at the "Investment Expenses". I don't understand how transfer fee is considered as gain. When I see this data:
- BTC In-Out row 24 on Full Report
- Investment Expenses row 8 on FIFO Tax Report

The way I see it is that I paid fee to the Blockchain network for 0.01. I don't own this 0.01 anymore. Could you explain why this is taxed? Sorry I'm not an expert in tax related matters.

I.e, say I am a dropshipper. I buy a doll from A for $100 with shipping fee to my warehouse for $1, then someone buys my doll for $110. I send that person this doll for $1 shipping fee. Doesn't that mean that my gain is.

$110 - ($100 + $1 + $1) = $8

But in the crypto Investment Expenses above, it seems that I am actually gaining the transfer fee?

**Comment by eprbell** [comment]
The transfer fee affects the in/out relationship (because it's crypto that is coming out of your stash) and as such they need to be part of tax calculation, so RP2 captures them in their own "Investment expense" category. However RP2 doesn't tell users what is taxed/not taxed on any given year: it performs lot fractioning and in/out lot pairing (according to the selected accounting method), and then it groups all lot fractions in categories (corresponding to each sheet in the tax_report_us output). How to treat each category from the tax perspective is the user's responsibility: if you're not 100% sure what to do, speaking with a tax professional is a good idea.

In the specific case of the transfer fee I heard two interpretations:
a) it's a non-taxable investment expense;
b) it's crypto that is coming out of your stash, so it's similar to a sale and therefore it needs to be taxed.
If anybody has a definitive answer to this quesiton (with a source) I can add it to the FAQ.

If you use a) then you can ignore the investment expense tab altogether, otherwise you have to use it to file taxes. 

To summarize, RP2 does all the calculations and groups all possible taxable events, but it doesn't tell you what actually needs to be filed or what forms to use (that can change from year to year).

Hope this helps.

**Comment by swissarmybox** [comment]
> The transfer fee affects the in/out relationship (because it's crypto that is coming out of your stash) and as such they need to be part of tax calculation, so RP2 captures them in their own "Investment expense" category. However RP2 doesn't tell users what is taxed/not taxed on any given year: it performs lot fractioning and in/out lot pairing (according to the selected accounting method), and then it groups all lot fractions in categories (corresponding to each sheet in the tax_report_us output). How to treat each category from the tax perspective is the user's responsibility: if you're not 100% sure what to do, speaking with a tax professional is a good idea.
> 
> In the specific case of the transfer fee I heard two interpretations: a) it's a non-taxable investment expense; b) it's crypto that is coming out of your stash, so it's similar to a sale and therefore it needs to be taxed. If anybody has a definitive answer to this quesiton (with a source) I can add it to the FAQ.
> 
> If you use a) then you can ignore the investment expense tab altogether, otherwise you have to use it to file taxes.
> 
> To summarize, RP2 does all the calculations and groups all possible taxable events, but it doesn't tell you what actually needs to be filed or what forms to use (that can change from year to year).
> 
> Hope this helps.

Oh I see, okay I think I got it confused. That makes sense. Thanks.

As for the investment expense. This seems to be the source.
https://www.blog.cointracking.info/is-transferring-bitcoin-between-wallets-taxable

It seems that it is indeed an investment expense.

**Comment by eprbell** [comment]
No, you're asking good questions and it's good to discuss publicly so that others can read too. Also thanks for the link.
